.github/workflows/test-lang-java.yml

```diff
@@ -35,52 +35,55 @@ concurrency:
 
 jobs:
   test:
-    name: Java ${{ matrix.java }} Test
+    name: 'Java Test'
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        java:
-        - '8'
-        - '11'
-        - '17'
-        - '21'
-        - '22-ea'
     steps:
-      - uses: actions/checkout@v4
+      - name: 'Checkout sourcecode'
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
 
-      - name: Cache Local Maven Repository
-        uses: actions/cache@v4
+      - name: 'Cache Local Maven Repository'
+        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
```

Comment:
Why do we need to go from v4 to v3.3.1?

Reply:
Oops

Reply:
I changed this when the 4.0.0 waas not yet released. Then later did a rebase onto main where dependabot had updated this.